addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/handler/KonnectedHandler.java

```diff
@@ -0,0 +1,330 @@
+     * it adds the appropriate senors and actuators
+     * as well as the location of the callback servelet
+     * and auth_token which can be used for validation
+     *
+     */
+    private String contructSettingsPayload() {
+        String hostPath = "";
+        try {
+            hostPath = getHostName() + webHookServlet.getPath();
+            logger.debug("The host path is: {}", hostPath);
+        }
+
+        catch (UnknownHostException e) {
+            logger.debug("Unable to obtain hostname: {}", e);
+            return "none";
+        }
+
+        String authToken = config.Auth_Token;
+        logger.debug("The Auth_Token is: {}", authToken);
+        logger.debug("The Sensor String is: {}", sensors.toString());
+        logger.debug("The Actuator String is: {}", actuators.toString());
+
+        updatepinstate();
+        String payload = "{\"sensors\":" + sensors.toString() + ",\"actuators\": " + actuators.toString()
+                + ",\"token\": \"" + authToken + "\",\"apiUrl\": \"http://" + hostPath + "\"}";
+        logger.debug("The payload is: {}", payload);
+        return payload;
+    }
+
+    protected String getHostName() throws UnknownHostException {
+        // returns the local address of the openHAB server
+        InetAddress addr = InetAddress.getLocalHost();
+        String hostname = addr.getHostAddress() + ":8080";
+        return hostname;
+    }
+
+    /*
+     * this method sends the payload to the putsettings timer
+     * where it can then be sent to the module
+     * the actual request is held off for 30 seconds and is reset if
+     * another instance requests new settings due to an update
+     */
+    private void updateKonnectedModule() {
+        Map<String, String> properties = this.thing.getProperties();
+        String host = properties.get("ipAddress");
+        String payload = contructSettingsPayload();
+        // setting it up to wait 30 seconds before sending the put request
+        logger.debug("creating new timer");
+        putSettingsTimer = putSettingsTimer.startTimer(host + "/settings", payload);
+    }
+
+    // this method sets the list of boolean variables of which pins are actuators
+    private void setisAct() {
+        config = getConfigAs(KonnectedConfiguration.class);
+
+        isAct.add(0, false);
+        isAct.add(1, config.isAct1);
+        isAct.add(2, config.isAct2);
+        isAct.add(3, config.isAct3);
+        isAct.add(4, config.isAct4);
+        isAct.add(5, config.isAct5);
+        // pins 6 can never be an actuator and pin 7 is always out
+        // so we set them accordingly for all things
+        isAct.add(6, false);
+        isAct.add(7, true);
+    }
+
+    /*
+     * This method ensures that the pins that are in the sensor/actuator
+     * lists match the boolean types the user has set
+     * it will remove any that don't match
+     */
+    private void updatepinstate() {
+        for (int i = 1; i < 6; i++) {
```

Comment:
The [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) 6 should be turned into a constant so people know what it stands for.